### PR TITLE
Upgrade wasmvm to v1.0.0-beta9; use libwasmvm_muslc.aarch64.a in docker

### PR DESCRIPTION
**Pull request by webmaster128**
This is woking now 🎉

Steps to test:
- Checkup this branch
- Build x86_64 image: `docker build --pull -t "cosmwasm/wasmd:manual" .` (unchanged)
- Build aarch64 image: `docker build --pull -t "cosmwasm/wasmd:manual" --build-arg arch=aarch64 .`

### CODE REVIEW

**Review by alpe — APPROVED** [review]
This is a milestone 🏆 and will make many devs happy.

I have built and manually tested on OSX Intel but it would be good to have @maurolacy to also confirm for M1. 
With this working, let's remove ` M1 macs are not fully supported. ` in https://github.com/CosmWasm/wasmd#supported-systems

**Comment by maurolacy** [comment]
I'm getting
```
#19 73.08 /usr/lib/gcc/aarch64-alpine-linux-musl/10.3.1/../../../../aarch64-alpine-linux-musl/bin/ld: skipping incompatible /lib/../lib/libwasmvm_muslc.a when searching for -lwasmvm_muslc
#19 73.08 /usr/lib/gcc/aarch64-alpine-linux-musl/10.3.1/../../../../aarch64-alpine-linux-musl/bin/ld: skipping incompatible /lib/libwasmvm_muslc.a when searching for -lwasmvm_muslc
#19 73.08 /usr/lib/gcc/aarch64-alpine-linux-musl/10.3.1/../../../../aarch64-alpine-linux-musl/bin/ld: cannot find -lwasmvm_muslc
#19 73.08 collect2: error: ld returned 1 exit status
#19 96.30 make: *** [Makefile:85: build] Error 2
```
when building for x86_64. (`docker build --pull -t "cosmwasm/wasmd:manual" .`)

The arm64 build worked.

**Comment by webmaster128** [comment]
> when building for x86_64. (`docker build --pull -t "cosmwasm/wasmd:manual" .`)

This is expected since you start with an ARM base image when working on an M1 machine. There are two options to build thimages now:
1. Build each arch on a matching host natively
2. Use docker cross arch building

This is going to be the next question for the deployment: do we build the ARM version in an ARM machine on CircleCI (1.)? Or do we use cross builds and build both from the Intel machines (2.)?

**Comment by maurolacy** [comment]
Makes sense. In fact, I now tried with

```
docker buildx build --platform linux/amd64 --pull -t "cosmwasm/wasmd:manual" .
```

and it worked. It takes ~9 minutes to build the amd64 image this way, though.

**Comment by alpe** [comment]
> This is going to be the next question for the deployment: do we build the ARM version in an ARM machine on CircleCI (1.)? Or do we use cross builds and build both from the Intel machines (2.)?

We do not build release artifacts with wasmd but leave this to the implementing chains.

**Comment by webmaster128** [comment]
> We do not build release artifacts with wasmd but leave this to the implementing chains.

I'm thinking of https://hub.docker.com/r/cosmwasm/wasmd, which is heavily used for development. I think it's worth considering to have both an x86_64 and and aarch64 build using docker's multi-arch functionality. But this is a separate ticket.

**Comment by ethanfrey** [comment]
>  I think it's worth considering to have both an x86_64 and and aarch64 build using docker's multi-arch functionality.

Yeah, I did some multi arch build for some of the tinygo deps when getting things working on arm64 and once you figure out the process (build them both, then make some linking image that references them), it was actually relatively easy. Just a bit of a time sink the first time.

**Comment by webmaster128** [comment]
Thank you @maurolacy for the cross build instructions. I collected them in this wikipage for all 4 cases: https://github.com/CosmWasm/wasmd/wiki/Maintainer-notes

**Pull request merged**
